test/gtest/diagnostics.cpp

```diff
@@ -243,8 +257,18 @@ TEST_F(Diagnostics, CompilerOptions) {
         // treated as an argument to `--Wdisable`, then
         // `uninitialized_out_param` would still be enabled and a warning would
         // fire.
-        EXPECT_EQ(0u, ::P4::diagnosticCount());
+        EXPECT_EQ(0u, diagnosticCount());
     }
 }
 
+TEST_F(Diagnostics, BasicInfo) {
+    AutoCompileContext autoContext(new GTestContext);
+    info(P4::ErrorType::INFO_INFERRED, "test");
+    EXPECT_EQ(1u, diagnosticCount());
+    EXPECT_EQ(1u, infoCount());
+    EXPECT_EQ(0u, warningCount());
+    EXPECT_EQ(0u, errorCount());
+}
+
+
```

Comment:
Nit: I'm not sure if we have style guide on this. But is it better to leave a single blank line here?